src/Shared/ServerInfrastructure/StringUtilities.cs

```diff
@@ -418,6 +529,36 @@ private static bool CheckBytesInAsciiRange(Vector<sbyte> check)
             return Vector.GreaterThanAll(check, Vector<sbyte>.Zero);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool CheckBytesInAsciiRange(Vector256<sbyte> check, Vector256<sbyte> zero)
+        {
+            if (Avx2.IsSupported)
```

Comment:
It's not actually possible for this to be called without Avx2.IsSupported being true, right?

Reply:
Correct.

Right now I ask myself why I didn't go with a `Debug.Assert` to make this obvious.
There must be some other code pattern that influenced me to write it that way, but I can't remember.

Also why this method is `public` and not `private`. 

I'll update this to the be simpler.

Reply:
Ahhh, I didn't remember my own comment :wink:

https://github.com/dotnet/aspnetcore/pull/17556#discussion_r353360875
That's why it's written that way.

Which way to go? With `Debug.Assert` or leave as is (plus add a comment explaining this).